sos/doi/src/main/java/se/inera/intyg/common/doi/v1/model/converter/certificate/question/QuestionTerminalDodsorsakFoljdAv.java

```diff
@@ -149,22 +149,34 @@ private static int getElementId(String questionId) {
         }
     }
 
-
     public static List<Dodsorsak> toInternal(Certificate certificate, List<String> questionIds) {
-        List<Dodsorsak> dodsorsakList = new ArrayList<>();
-        for (String questionId : questionIds) {
-            if (certificate.getData().get(questionId) != null) {
-                final var terminalCauseOfDeath = (CertificateDataValueCauseOfDeath) certificate.getData().get(questionId)
-                    .getValue();
-                final var description = terminalCauseOfDeath.getDescription().getText();
-                final var debut = terminalCauseOfDeath.getDebut().getDate() != null
-                    ? new InternalDate(terminalCauseOfDeath.getDebut().getDate()) : null;
-                final var specifikation = terminalCauseOfDeath.getSpecification().getCode() != null
-                    ? Specifikation.fromValue(terminalCauseOfDeath.getSpecification().getCode()) : null;
+        final List<Dodsorsak> dodsorsakList = questionIds.stream()
```

Comment:
Is it possible to break this into smaller chunks or private methods to make this more descriptive?

Reply:
Sure, just used the original code. Done!